source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_helper.py

```diff
@@ -66,13 +68,143 @@ def build_ridesharing_crowfly_journey(instance, origin, destination, period_extr
+            if rsj.driver.rate_count:
+                rs_section.ridesharing_information.driver.rating.count = rsj.driver.rate_count
+            if rsj.metadata.rating_scale_min is not None and rsj.metadata.rating_scale_max is not None:
+                rs_section.ridesharing_information.driver.rating.scale_min = rsj.metadata.rating_scale_min
+                rs_section.ridesharing_information.driver.rating.scale_max = rsj.metadata.rating_scale_max
+
+            # TODO links
+
+            # TODO CO2 = length * coeffCar / (totalSeats  + 1)
+            rs_section.length = rsj.distance
+            rs_section.shape.extend([pickup_coord, dropoff_coord]) # TODO real shape
+            rs_section.duration = rsj.dropoff_date_time - rsj.pickup_date_time
+            rs_section.begin_date_time = rsj.pickup_date_time
+            rs_section.end_date_time = rsj.dropoff_date_time
+            # report values to journey
+            pb_rsj.distances.ridesharing += rs_section.length
+            pb_rsj.duration += rs_section.duration
+            pb_rsj.durations.total += rs_section.duration
+            pb_rsj.durations.ridesharing += rs_section.duration
+
+            # end teleport section
+            end_teleport_section = pb_rsj.sections.add()
+            end_teleport_section.id = "section_{}".format(six.text_type(generate_id()))
+            end_teleport_section.type = response_pb2.CROW_FLY
+            end_teleport_section.street_network.mode = response_pb2.Walking
+            end_teleport_section.origin.CopyFrom(pb_rsj_dropoff)
+            end_teleport_section.destination.CopyFrom(to_pt_obj)
+            end_teleport_section.length = int(crowfly_distance_between(dropoff_coord, to_coord))
+            end_teleport_section.duration = 0
+            end_teleport_section.shape.extend([dropoff_coord, to_coord])
+            end_teleport_section.begin_date_time = rsj.dropoff_date_time
+            end_teleport_section.end_date_time = rsj.dropoff_date_time
+            # report value to journey
+            pb_rsj.distances.walking += end_teleport_section.length
+
+            # create ticket associated
+            ticket = response_pb2.Ticket()
+            ticket.id = "ticket_{}".format(six.text_type(generate_id()))
+            ticket.name = "ridesharing_price_{}".format(ticket.id)
+            ticket.found = True
+            ticket.comment = "Ridesharing price for section {}".format(rs_section.id)
+            ticket.section_id.extend([rs_section.id])
+            # also add fare to journey
+            ticket.cost.value = rsj.price
+            pb_rsj.fare.total.value = ticket.cost.value
+            if rsj.currency == "EUR":
+                ticket.cost.currency = "centime"
+                pb_rsj.fare.total.currency = ticket.cost.currency
+            pb_rsj.fare.found = True
+            pb_rsj.fare.ticket_id.extend([ticket.id])
+
+            pb_tickets.append(ticket)
+            pb_rsjs.append(pb_rsj)
+
+        except Exception as e:
```

Comment:
it's a bit too much. 
If you have some specific exception to catch you can do it specifically, but we shouldn't catch everything everywhere.
We need to have *one* place where we do a ```except:``` to prevent error in ridesharing to propagate but it should be near top level, so in the manager or in the calling scenario, that's all.

Reply:
this exception can be removed